instrumentation/opentelemetry/internal/metrics/system_metrics.go

```diff
@@ -0,0 +1,119 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/tklauser/go-sysconf"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const meterName = "hypertrace.goagent.metrics"
+
+type systemMetrics struct {
+	memory          float64
+	cpuSecondsTotal float64
+}
+
+type processStats struct {
+	utime  float64
+	stime  float64
+	cutime float64
+	cstime float64
+	rss    float64
+}
+
+const procStatArrayLength = 52
+
+var (
+	clkTck   = getClockTicks()
+	pageSize = float64(os.Getpagesize())
+)
+
+func InitializeSystemMetrics() {
+	meterProvider := otel.GetMeterProvider()
+	meter := meterProvider.Meter(meterName)
+	err := setUpMetricRecorder(meter)
+	if err != nil {
+		fmt.Println("error initialising metrics, failed to setup metric recorder")
```

Comment:
please remove fmt based prints? I hope there should be a logger available in goagent.

Reply:
We just use golangs `log`. There is no zap available. @thugrock7 eg. `log.Printf("error while shutting down tracer provider: %v\n", err)`